src/Sylius/Bundle/CoreBundle/Validator/Constraints/OrderShippingMethodEligibilityValidator.php

```diff
@@ -42,9 +43,21 @@ public function validate($value, Constraint $constraint): void
         }
 
         foreach ($shipments as $shipment) {
-            if (!$this->methodEligibilityChecker->isEligible($shipment, $shipment->getMethod())) {
+            /** @var ShippingMethodInterface $shippingMethod */
+            $shippingMethod = $shipment->getMethod();
+
+            if (!$shippingMethod->isEnabled() || !$shippingMethod->getChannels()->contains($value->getChannel())) {
+                $this->context->addViolation(
+                    $constraint->getMethodNotAvailableMessage(),
+                    ['%shippingMethodName%' => $shippingMethod->getName()],
+                );
+
+                continue;
+            }
```

Comment:
I've suggested putting this logic in the new shipping eligibility checker instead of validator 🤔 However, there would be a problem with the message which could be misleading